python/ray/serve/controller.py

```diff
@@ -973,9 +914,7 @@ def deploy_serve_application(
             app.deployments[unique_deployment_name].set_options(
                 **options, _internal=True
             )
-
-        # Run the application locally on the cluster.
-        serve.run(app, name=name, route_prefix=route_prefix)
+        return app.deploy_args_list
```

Comment:
why not return app directly?

Reply:
Deployment config and replica config need to be serialized: https://github.com/ray-project/ray/pull/34117/files#diff-0b24164442506c685948f2a5433516040ebcd834917466c8e60df73ad3a0ee51R73-R74.

I think some parts of the code path might be able to be simplified further, but my attempt to do so brought me down a rabbit hole so we can iterate on this in future PRs once the overall architecture change is finished in this PR.